src/backend/joanie/core/serializers/admin.py

```diff
@@ -899,6 +935,46 @@ class Meta:
         read_only_fields = ["id", "course", "course_runs", "is_graded", "position"]
 
 
+class AdminTeacherSerializer(serializers.ModelSerializer):
+    """Serializer for Teacher model."""
+
+    class Meta:
+        model = models.Teacher
+        fields = ["id", "first_name", "last_name"]
+        read_only_fields = ["id"]
+
+
+class AdminSkillSerializer(serializers.ModelSerializer):
+    """Serializer for Skill model."""
+
+    title = serializers.CharField()
+
+    class Meta:
+        model = models.Skill
+        fields = ["id", "title"]
+        read_only_fields = ["id"]
+
+    def validate(self, attrs):
+        """
+        Validate that the skill has a product provided then bind the product instance
+        to validated_data until serializer instance accepts partial data.
+        """
+        validated_data = super().validate(attrs)
+        product_id = self.initial_data.get("product_id", None)
+
+        if product_id:
+            try:
+                validated_data["products"] = models.Product.objects.filter(
+                    id=product_id
+                )
+            except models.Course.DoesNotExist as exception:
+                raise serializers.ValidationError(
+                    {"course_id": "Resource {course_id} does not exist."}, exception
```

Comment:
What it supposed to be an **f string** ? or am i missing something... where is `course_id` variable ?